File: Unidad_2/04_Lambda_con_ambiguedad/Principal.java
public class Principal{
	public static void main(String[] args){
		//Solucion 1
		System.out.println("\tSolucion 1");
		Principal.engine((long x, long y) -> x + y);
		Principal.engine((int x, int y) -> x * y);
		Principal.engine((long x, long y) -> x / y);

		//Solucion 2
		System.out.println("\tSolucion 2");
		Principal.engine((CalculadoraInt)(x, y) -> x - y);
		Principal.engine((CalculadoraLong)(x, y) -> x % y);

		//Solucion 3
		System.out.println("\tSolucion 3");
		CalculadoraLong _calLong = (x, y) -> x + y;
		CalculadoraInt _calInt = (x, y) -> x * y;
		Principal.engine(_calLong);
		Principal.engine(_calInt);
	}

	// Sobre carga de metodos
	private static void engine(CalculadoraInt cal){
		int x = 2, y = 4;
		int resultado = cal.calculate(x, y);
		System.out.println("CalculadoraInt ==> resultado =" + resultado);
	}

	private static void engine(CalculadoraLong cal){
		long x = 2, y = 4;
		long resultado =  cal.calculate(x, y);
		System.out.println("CalculadoraLong ==> resultado =" + resultado);
	}
}

File: 2 _oportunidad/_07_ejercicios/retorno_lambda_v2/Principal.py
#mi solucion --> correcto
class Principal:
    @staticmethod
    def main(args):
        Principal.engine(lambda x, y :lambda x1, y2 : x1+y2)
    @staticmethod
    def _engine(cal):
        x =2
        y =4
        resultado = cal.calcular(x,y)
        print(resultado.calcular(int(x), int(y)))
    @staticmethod
    def _engine(cal):
        x =2
        y =4
        resultado = cal.calcular(x,y)
        print(resultado)
